File: 500-keyboard-row/keyboard-row.py
class Solution:
    def findWords(self, words):
        row1 = set("qwertyuiop")
        row2 = set("asdfghjkl")
        row3 = set("zxcvbnm")

        ans = []

        for word in words:
            if (
                all(ch.lower() in row1 for ch in word)
                or all(ch.lower() in row2 for ch in word)
                or all(ch.lower() in row3 for ch in word)
            ):
                ans.append(word)

        return ans

File: 500-keyboard-row/keyboard-row.java
import java.util.*;

class Solution {
    public String[] findWords(String[] words) {

        String row1 = "qwertyuiop";
        String row2 = "asdfghjkl";
        String row3 = "zxcvbnm";

        List<String> ans = new ArrayList<>();

        for (String word : words) {

            String lower = word.toLowerCase();

            String row;

            if (row1.indexOf(lower.charAt(0)) != -1) {
                row = row1;
            }
            else if (row2.indexOf(lower.charAt(0)) != -1) {
                row = row2;
            }
            else {
                row = row3;
            }

            boolean valid = true;

            for (char ch : lower.toCharArray()) {

                if (row.indexOf(ch) == -1) {
                    valid = false;
                    break;
                }
            }

            if (valid) {
                ans.add(word);
            }
        }

        return ans.toArray(new String[0]);
    }
}